larray/viewer.py

```diff
@@ -1983,11 +1983,13 @@ def setup_and_check(self, data, title='', readonly=False, minvalue=None, maxvalu
         if icon is not None:
             self.setWindowIcon(icon)
 
-        if not title:
-            title = _("Session viewer") if readonly else _("Session editor")
+        if title:
+            self.title = title
+        else:
+            self.title = _("Session viewer") if readonly else _("Session editor")
         if readonly:
-            title += ' (' + _('read only') + ')'
-        self.setWindowTitle(title)
+            self.title += ' (' + _('read only') + ')'
+        self.setWindowTitle(self.title)
```

Comment:
in this kind of case, I prefer to use simply "title" and set self.title at the end.